### DISCUSSION TITLE
Next ServersideProps returns mapping is not function

### DISCUSSION THREAD

**Discussion by akhilpamwani** [vercel/next.js#36450] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

For my web app

For the frontend

{
  "name": "hunting-coder",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "aos": "^2.3.4",
    "axios": "^0.26.0",
    "next": "^12.1.5",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-router-dom": "^6.3.0",
    "react-toastify": "^8.2.0",
    "sass": "^1.49.7",
    "sharp": "^0.30.1"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.2",
    "eslint": "8.8.0",
    "eslint-config-next": "12.0.10",
    "postcss": "^8.4.6",
    "tailwindcss": "^3.0.22"
  }
}

For the backend

{
  "dependencies": {
    "bcrypt": "^5.0.1",
    "body-parser": "^1.19.2",
    "cors": "^2.8.5",
    "dotenv": "^16.0.0",
    "express": "^4.17.3",
    "express-jwt": "^6.1.1",
    "jsonwebtoken": "^8.5.1",
    "mongoose": "^6.2.3",
    "nodemailer": "^6.7.3",
    "nodemailer-sendgrid": "^1.0.3",
    "nodemon": "^2.0.15",
    "slugify": "^1.6.5"
  },
  "name": "backend",
  "version": "1.0.0",
  "main": "server.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "server": "nodemon server"
  },
  "author": "",
  "license": "ISC",
  "description": ""
}

### What browser are you using? (if relevant)

Microsoft  Edge

### How are you deploying your application? (if relevant)

Now it is under development

### Describe the Bug

There is problem in getServersideprops when i try to fetch data from my express server.It returns TypeError: Cannot read properties of undefined (reading 'map'). This is my github repo :-  

For the frontend :-

https://github.com/akhilpamwani/CodeHustler

For the  backend :-

https://github.com/akhilpamwani/CodeHustler-Backend-

### Expected Behavior

After starting server TypeError: Cannot read properties of undefined (reading 'map')
![image](https://user-images.githubusercontent.com/83491855/165117968-08937ea0-7433-49e3-a41c-340e65cbbe49.png)

### To Reproduce

1  Clone my repo from  https://github.com/akhilpamwani/CodeHustler-Backend- and https://github.com/akhilpamwani/CodeHustler.
2 Start both the server next app yarn dev and express server yarn server.
3 You will the the mapping error.

**Comment by ijjk** [comment]
Hi, it looks like the code shown in the screenshot is not present in [the repo here](https://github.com/akhilpamwani/CodeHustler/blob/dcd0884613911a466bb4202e55925f342d3a5f79/Components/Blogs/Blogs.js). 

Note, if you are adding `getServerSideProps` in that file it will not work as `getServerSideProps` can only be used in files in your `pages` directory.

**Reply by akhilpamwani** [reply]
Thanks ! It  worked
